Nightly build errors with cuda/11.{1,2}+gcc/{8,9}.3.0 with -Werror: error: attribute ignored with [[nodiscard]]

**Describe the bug**
Nightly cuda/11.1+gcc/8.3.0 builds fail to compile with -Werror following merge of #6320 with output:
```
20:33:35 /data/jenkins-new/workspace/kokkos_semsrhel7gpu01_cuda111_gcc830_cpp17/kokkos/simd/src/Kokkos_SIMD_Scalar.hpp:138:1: error: attribute ignored [-Werror=attributes]
20:33:35    [[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION friend constexpr simd operator*(
20:33:35  ^
20:33:35 /data/jenkins-new/workspace/kokkos_semsrhel7gpu01_cuda111_gcc830_cpp17/kokkos/simd/src/Kokkos_SIMD_Scalar.hpp:138:1: note: an attribute that appertains to a type-specifier is ignored
20:33:35 /data/jenkins-new/workspace/kokkos_semsrhel7gpu01_cuda111_gcc830_cpp17/kokkos/simd/src/Kokkos_SIMD_Scalar.hpp:142:1: error: attribute ignored [-Werror=attributes]
20:33:35    [[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION friend constexpr simd operator/(
20:33:35  ^
20:33:35 /data/jenkins-new/workspace/kokkos_semsrhel7gpu01_cuda111_gcc830_cpp17/kokkos/simd/src/Kokkos_SIMD_Scalar.hpp:142:1: note: an attribute that appertains to a type-specifier is ignored
20:33:35 /data/jenkins-new/workspace/kokkos_semsrhel7gpu01_cuda111_gcc830_cpp17/kokkos/simd/src/Kokkos_SIMD_Scalar.hpp:146:1: error: attribute ignored [-Werror=attributes]
20:33:35    [[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION friend constexpr simd operator+(
20:33:35  ^
20:33:35 /data/jenkins-new/workspace/kokkos_semsrhel7gpu01_cuda111_gcc830_cpp17/kokkos/simd/src/Kokkos_SIMD_Scalar.hpp:146:1: note: an attribute that appertains to a type-specifier is ignored
20:33:35 /data/jenkins-new/workspace/kokkos_semsrhel7gpu01_cuda111_gcc830_cpp17/kokkos/simd/src/Kokkos_SIMD_Scalar.hpp:150:1: error: attribute ignored [-Werror=attributes]
20:33:35    [[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION friend constexpr simd operator-(
20:33:35  ^
20:33:35 /data/jenkins-new/workspace/kokkos_semsrhel7gpu01_cuda111_gcc830_cpp17/kokkos/simd/src/Kokkos_SIMD_Scalar.hpp:150:1: note: an attribute that appertains to a type-specifier is ignored
20:33:35 /data/jenkins-new/workspace/kokkos_semsrhel7gpu01_cuda111_gcc830_cpp17/kokkos/simd/src/Kokkos_SIMD_Scalar.hpp:154:1: error: attribute ignored [-Werror=attributes]
20:33:35    [[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION friend constexpr simd operator>>(
...
```

**Please also include the following items to support reproducing the bug**
**Reproducer** (kokkos-dev-2 or ):
```bash
module load sems-cmake sems-gcc/8.3.0 sems-cuda/11.1.0

$KOKKOS_PATH/generate_makefile.bash --with-devices=Cuda,Serial --arch=Volta70 --compiler=$KOKKOS_PATH/bin/nvcc_wrapper --cxxflags="-O3 -Wall -Wunused-parameter -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wuninitialized " --cxxstandard="17" --kokkos-path=$KOKKOS_PATH --with-cuda-options=enable_lambda  --no-examples   --with-options=enable_large_mem_tests
```

--- a/simd/src/Kokkos_SIMD.hpp
+++ b/simd/src/Kokkos_SIMD.hpp
@@ -19,6 +19,13 @@
 
 #include <Kokkos_SIMD_Common.hpp>
 
+// suppress NVCC warnings with the [[nodiscard]] attribute on overloaded
+// operators implemented as hidden friends
+#if defined(KOKKOS_COMPILER_NVCC) && KOKKOS_COMPILER_NVCC < 1130
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wattributes"
+#endif
+
 #include <Kokkos_SIMD_Scalar.hpp>
 
 #ifdef KOKKOS_ARCH_AVX2
@@ -31,6 +38,10 @@
 
 #ifdef __ARM_NEON
 #include <Kokkos_SIMD_NEON.hpp>
+#endif
+
+#if defined(KOKKOS_COMPILER_NVCC) && KOKKOS_COMPILER_NVCC < 1130
+#pragma GCC diagnostic pop
 #endif
 
 namespace Kokkos {

